sql/core/src/main/scala/org/apache/spark/sql/execution/SparkSqlParser.scala

```diff
@@ -90,6 +93,51 @@ class SparkSqlAstBuilder(conf: SQLConf) extends AstBuilder(conf) {
     ResetCommand
   }
 
+  /**
+   * Create a [[SetCommand]] logical plan to set [[SQLConf.SESSION_LOCAL_TIMEZONE]]
+   * Example SQL :
+   * {{{
+   *   SET TIME ZONE LOCAL;
+   *   SET TIME ZONE 'Asia/Shanghai';
+   *   SET TIME ZONE INTERVAL 10 HOURS;
+   * }}}
+   */
+  override def visitSetTimeZone(ctx: SetTimeZoneContext): LogicalPlan = withOrigin(ctx) {
+    val key = SQLConf.SESSION_LOCAL_TIMEZONE.key
+    if (ctx.interval != null) {
+      val interval = parseIntervalLiteral(ctx.interval)
+      if (interval.months != 0 || interval.days != 0 ||
+        math.abs(interval.microseconds) > 18 * DateTimeConstants.MICROS_PER_HOUR) {
+        throw new ParseException("The interval value must be in the range of [-18, +18] hours",
```

Comment:
shall we also fail for things like `INTERVAL 1 MICROSECOND`?

Reply:
If we want to fail this like this completely, we need a new interval parser, otherwise, we can only forbid it based on the interval result. e.g. `INTERVAL 1 MICROSECOND  -1 MICROSECOND 1 HOUR` will work...

Reply:
`INTERVAL 1 MICROSECOND -1 MICROSECOND 1 HOUR` is OK. We should forbid truncation like `INTERVAL 1 MICROSECOND`, which you have to ignore the 1 microsecond.